gamersnet_backend/routes/posts/getPosts.js

```diff
@@ -1,10 +1,51 @@
 'use strict';
 
 
-// include our function from the database to add post
-let {getAllPosts, getValidPosts} = require('../../persistence/posts');
+// include our function from the database to fetch posts
+let {getUserPosts, getAllPosts, getValidPosts} = require('../../persistence/posts');
 
-// this function handles the /post/getAllPosts/ endpoint
+let {verifyUserLoggedIn} = require('../utilities/tokenUtility')
+
+let {getUserIDFromToken} = require('../../persistence/tokens.js')
+
+// this function handles the /posts/listUserPosts/ endpoint
+async function listUserPosts(request, response) {
+    let body = request.body;
+    let cookie = request.headers.cookie;
+
+    let loggedIn = false;
+
+    if (cookie) {
+        cookie = cookie.split('=')[1];
+        loggedIn = await verifyUserLoggedIn(cookie);
+    }
+
+    if (loggedIn){
+        let tokenDocument = await getUserIDFromToken(cookie);
+
+        // Note: when fetching users' posts, an empty string is passed if it is requested by the user that's logged in
+        // This can/should be changed later but for now , it's okay
+        // - Jay
+        let userID = body.userID;
+        if(userID == ""){
+            userID = tokenDocument.userID;
+        }
```

Comment:
Considering how this fits the description of a `GET`, shouldn't the user's ID be in the URL rather than in the body?

Reply:
You are right! I didn't think about it!